public/app/features/alerting/unified/components/rules/central-state-history/CentralAlertHistory.tsx

```diff
@@ -0,0 +1,368 @@
+  ({ showClearFilterSuffix, onClearFilterClick, ...rest }: SearchFieldInputProps, ref) => {
+    const placeholder = t('central-alert-history.filter.placeholder', 'Filter events in the list by labels');
+    return (
+      <Field
+        label={
+          <Label htmlFor="eventsSearchInput">
+            <Stack gap={0.5}>
+              <span>
+                <Trans i18nKey="central-alert-history.filter.label">Filter events</Trans>
+              </span>
+            </Stack>
+          </Label>
+        }
+      >
+        <Input
+          id="eventsSearchInput"
+          prefix={<Icon name="search" />}
+          suffix={
+            showClearFilterSuffix && (
+              <Button fill="text" icon="times" size="sm" onClick={onClearFilterClick}>
+                <Trans i18nKey="central-alert-history.filter.button.clear"> Clear</Trans>
+              </Button>
+            )
+          }
+          placeholder={placeholder}
+          ref={ref}
+          {...rest}
+        />
+      </Field>
+    );
+  }
+);
+
+SearchFieldInput.displayName = 'SearchFieldInput';
+
+function EventRow({ record }: { record: LogRecord }) {
+  const styles = useStyles2(getStyles);
+  const [isCollapsed, setIsCollapsed] = useState(true);
+  return (
+    <div>
+      <div className={styles.header} data-testid="rule-group-header">
+        <CollapseToggle
+          size="sm"
+          className={styles.collapseToggle}
+          isCollapsed={isCollapsed}
+          onToggle={setIsCollapsed}
+        />
+        <Stack gap={0.5} direction={'row'} alignItems={'center'}>
+          <div className={styles.timeCol}>
+            <Timestamp time={record.timestamp} />
+          </div>
+          <div className={styles.transitionCol}>
+            <EventTransition previous={record.line.previous} current={record.line.current} />
+          </div>
+          <div className={styles.alertNameCol}>
+            {record.line.labels ? <AlertRuleName labels={record.line.labels} ruleUID={record.line.ruleUID} /> : null}
+          </div>
+          <div className={styles.labelsCol}>
+            <AlertLabels labels={record.line.labels ?? {}} limit={LIMIT_LABELS} />
+          </div>
+        </Stack>
+      </div>
+    </div>
+  );
+}
+
+function AlertRuleName({ labels, ruleUID }: { labels: Record<string, string>; ruleUID?: string }) {
+  const styles = useStyles2(getStyles);
+  const alertRuleName = labels['alertname'];
+  if (!ruleUID) {
+    return <Text>{alertRuleName}</Text>;
+  }
+  return (
+    <a href={`/alerting/${GRAFANA_DATASOURCE_NAME}/${ruleUID}/view`} className={styles.alertName}>
+      {alertRuleName}
+    </a>
+  );
+}
+
+interface EventTransitionProps {
+  previous: GrafanaAlertStateWithReason;
+  current: GrafanaAlertStateWithReason;
+}
+function EventTransition({ previous, current }: EventTransitionProps) {
+  return (
+    <Stack gap={0.5} direction={'row'}>
+      <EventState state={previous} />
```

Comment:
These are probably not final, the design in the Figma file is a tad outdated – [this file](https://github.com/grafana/grafana/blob/main/public/app/features/alerting/unified/components/rule-list/RuleListIcon.tsx) has the updated icons for state / health (but doesn't include missing series).

Reply:
That could be great!, but I'm not sure if I can use this component , as in the Loki line we have `GrafanaAlertStateWithReason`, while in the RuleListIcon we have `PromAlertingRuleState`. Do you still think we can? 🤔
```
export enum PromAlertingRuleState {
  Firing = 'firing',
  Inactive = 'inactive',
  Pending = 'pending',
}

export enum GrafanaAlertState {
  Normal = 'Normal',
  Alerting = 'Alerting',
  Pending = 'Pending',
  NoData = 'NoData',
  Error = 'Error',
}

type GrafanaAlertStateReason = ` (${string})` | '';

export type GrafanaAlertStateWithReason = `${GrafanaAlertState}${GrafanaAlertStateReason}`;
```

Reply:
updated the icons!